R/make.R

```diff
@@ -355,18 +355,27 @@ graph.lattice <- function(dimvector = NULL , length = NULL , dim = NULL , nei =
     periodic <- circular
   }
 
+  if (is.numeric(length) && length != floor(length)) {
+    warning("length was rounded to the nearest integer")
+    length <- round(length)
+  }
+
   if (is.null(dimvector)) {
     dimvector <- rep(length, dim)
   }
 
+  if (length(periodic) == 1) {
+    periodic <- rep(periodic, length(dimvector))
+  }
+
   on.exit(.Call(R_igraph_finalizer))
   res <- square_lattice_impl(dimvector, nei, directed, mutual, periodic)
   if (igraph_opt("add.params")) {
     res$name <- "Lattice graph"
     res$dimvector <- dimvector
     res$nei <- nei
     res$mutual <- mutual
-    res$circular <- circular
+    res$circular <- periodic
```

Comment:
Perhaps:

```suggestion
    res$periodic <- periodic
    res$circular <- periodic
```

Reply:
we don't have this in the original body copied here.